hack/e2e/environment.sh

```diff
@@ -17,62 +32,68 @@ function os_check() {
     unset CAPM3_LOCAL_IMAGE
-    export M3PATH=${M3PATH:-"${HOME}/go/src/github.com/metal3-io"}
     export CAPM3PATH="${M3PATH}/cluster-api-provider-metal3"
 fi
-
-# needed for variable substitution in templates
-export IMAGE_CHECKSUM_TYPE="md5"
-
-# shellcheck disable=SC2016
-export CONTROL_PLANE_MACHINE_COUNT=${CONTROL_PLANE_MACHINE_COUNT:-3}
-# shellcheck disable=SC2016
-export WORKER_MACHINE_COUNT=${WORKER_MACHINE_COUNT:-1}
-
-# These two variables below are required to render the cluster template dynamically from the metal3-dev-env.
-# This template is used by default, but users can change to use the static templates by changing the flavor.
-export NUM_OF_CONTROLPLANE_REPLICAS=${CONTROL_PLANE_MACHINE_COUNT}
-export NUM_OF_WORKER_REPLICAS=${WORKER_MACHINE_COUNT}
-
-# The e2e test framework would itself handle the cloning. It clones all the repos that are cloned in M3-DEV-ENV expect CAPM3.
-# It would use the local CAPM3 repo where the e2e test is running.
-# Set this variable to false to avoid the dev-env to override what the test framework cloned.
-export FORCE_REPO_UPDATE="false"
-
-os_check
-
-if [[ "${OS}" == ubuntu ]]; then
-    export IMAGE_OS=${IMAGE_OS:-"ubuntu"}
+if [[ "${IMAGE_OS}" == "ubuntu" ]]; then
+  export UPGRADED_IMAGE_NAME="UBUNTU_20.04_NODE_IMAGE_K8S_${UPGRADED_K8S_VERSION}.qcow2"
+  export UPGRADED_RAW_IMAGE_NAME="UBUNTU_20.04_NODE_IMAGE_K8S_${UPGRADED_K8S_VERSION}-raw.img"
 else
-    export IMAGE_OS=${IMAGE_OS:-"centos"}
+  export UPGRADED_IMAGE_NAME="CENTOS_8_NODE_IMAGE_K8S_${UPGRADED_K8S_VERSION}.qcow2"
+  export UPGRADED_RAW_IMAGE_NAME="CENTOS_8_NODE_IMAGE_K8S_${UPGRADED_K8S_VERSION}-raw.img"
 fi
-M3_DEV_ENV_PATH="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-# shellcheck disable=SC1091
-# shellcheck disable=SC1090
-source "${M3_DEV_ENV_PATH}/scripts/feature_tests/feature_test_vars.sh"
-# shellcheck disable=SC1091
-# shellcheck disable=SC1090
-source "${M3_DEV_ENV_PATH}/scripts/feature_tests/node_reuse/node_reuse_vars.sh"
 
-# Pin Calico version
-export CALICO_PATCH_RELEASE="v3.21.0"
-export CALICO_MINOR_RELEASE="v3.21"
+# Exported to the cluster templates
+SSH_PUB_KEY_CONTENT=$(cat "$HOME/.ssh/id_rsa.pub")
+export SSH_PUB_KEY_CONTENT
+# The host that has images for provisioning, this should be in the
+# format of a URL host, e.g. with IPv6, it should be surrounded
+# by brackets
+export PROVISIONING_URL_HOST="172.22.0.1"
+export CLUSTER_PROVISIONING_IP="172.22.0.2"
+export CLUSTER_URL_HOST="$CLUSTER_PROVISIONING_IP"
+
+# Vars used in the tests 
+export IRONIC_NAMESPACE=${IRONIC_NAMESPACE:-"baremetal-operator-system"}
+export NAMEPREFIX=${NAMEPREFIX:-"baremetal-operator"}
+export IRONIC_DATA_DIR="$DEV_ENV_WORKING_DIR/ironic"
+export BMOPATH="${BMOPATH:-${M3PATH}/baremetal-operator}"
+export IRONIC_TLS_SETUP=${IRONIC_TLS_SETUP:-"true"}
+export IRONIC_BASIC_AUTH=${IRONIC_BASIC_AUTH:-"true"}
+export RESTART_CONTAINER_CERTIFICATE_UPDATED=${RESTART_CONTAINER_CERTIFICATE_UPDATED:-${IRONIC_TLS_SETUP}}
```

Comment:
Could we not set these in the go code or config file instead? Actually some of them are already in the e2e config.

Reply:
moved some to e2e config and others still needed in the var environment because they are used in other scrips